src/tokens/Assets.tsx

```diff
@@ -0,0 +1,425 @@
+import { headerWithBackButton } from 'src/navigator/Headers'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import useScrollAwareHeader from 'src/navigator/ScrollAwareHeader'
+import { StackParamList } from 'src/navigator/types'
+import {
+  positionsSelector,
+  positionsWithClaimableRewardsSelector,
+  totalPositionsBalanceUsdSelector,
+} from 'src/positions/selectors'
+import { Position } from 'src/positions/types'
+import { getFeatureGate } from 'src/statsig'
+import { StatsigFeatureGates } from 'src/statsig/types'
+import Colors from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { vibrateInformative } from 'src/styles/hapticFeedback'
+import { Shadow, Spacing, getShadowStyle } from 'src/styles/styles'
+import variables from 'src/styles/variables'
+import { PositionItem, TokenBalanceItem } from 'src/tokens/AssetItem'
+import {
+  stalePriceSelector,
+  tokensWithTokenBalanceSelector,
+  totalTokenBalanceSelector,
+} from 'src/tokens/selectors'
+import { TokenBalance } from 'src/tokens/slice'
+import { sortByUsdBalance } from 'src/tokens/utils'
+
+const DEVICE_WIDTH_BREAKPOINT = 340
+
+type Props = NativeStackScreenProps<StackParamList, Screens.Assets>
+interface SectionData {
+  appName?: string
+}
+
+const AnimatedSectionList = Animated.createAnimatedComponent<
+  SectionListProps<TokenBalance | Position, SectionData>
+>(SectionList<TokenBalance | Position, SectionData>)
+
+const assetIsPosition = (asset: Position | TokenBalance): asset is Position =>
+  'type' in asset && (asset.type === 'app-token' || asset.type === 'contract-position')
+
+export enum AssetTabType {
+  Assets = 0,
+  Collectibles = 1,
+  Positions = 2,
+}
+
+// offset relative to the bottom of the non sticky header component, where the
+// screen header opacity animation starts
+const HEADER_OPACITY_ANIMATION_START_OFFSET = 44
+// distance in points over which the screen header opacity animation is applied
+const HEADER_OPACITY_ANIMATION_DISTANCE = 20
+
+function AssetsScreen({ navigation, route }: Props) {
+  const { t } = useTranslation()
+
+  const activeTab = route.params?.activeTab ?? AssetTabType.Assets
+
+  const tokens = useSelector(tokensWithTokenBalanceSelector)
+  const localCurrencySymbol = useSelector(getLocalCurrencySymbol)
+  const totalTokenBalanceLocal = useSelector(totalTokenBalanceSelector) ?? new BigNumber(0)
+  const tokensAreStale = useSelector(stalePriceSelector)
+  const insets = useSafeAreaInsets()
+
+  const positions = useSelector(positionsSelector)
+  const showPositions = getFeatureGate(StatsigFeatureGates.SHOW_POSITIONS)
+  const displayPositions = showPositions && positions.length > 0
+
+  const dappShortcutsEnabled = getFeatureGate(StatsigFeatureGates.SHOW_CLAIM_SHORTCUTS)
+  const positionsWithClaimableRewards = useSelector(positionsWithClaimableRewardsSelector)
+  const showClaimRewards =
```

Comment:
Is this meant to render on _both_ the assets and dapp positions tabs?

Reply:
Confirmed that we want to keep this on the assets and dapp positions page